packages/block-editor/src/components/rich-text/index.js

```diff
@@ -163,49 +164,45 @@ export function RichTextWrapper(
 		isBlockSelected,
 	] );
 
-	const disableBoundBlocks = useSelect(
+	const { disableBoundBlock, bindingsPlaceholder } = useSelect(
 		( select ) => {
-			// Disable Rich Text editing if block bindings specify that.
-			let _disableBoundBlocks = false;
-			if ( blockBindings && canBindBlock( blockName ) ) {
-				const blockTypeAttributes =
-					getBlockType( blockName ).attributes;
-				const { getBlockBindingsSource } = unlock(
-					select( blocksStore )
-				);
-				for ( const [ attribute, binding ] of Object.entries(
-					blockBindings
-				) ) {
-					if (
-						blockTypeAttributes?.[ attribute ]?.source !==
-						'rich-text'
-					) {
-						break;
-					}
-
-					// If the source is not defined, or if its value of `canUserEditValue` is `false`, disable it.
-					const blockBindingsSource = getBlockBindingsSource(
-						binding.source
-					);
-					if (
-						! blockBindingsSource?.canUserEditValue?.( {
-							select,
-							context: blockContext,
-							args: binding.args,
-						} )
-					) {
-						_disableBoundBlocks = true;
-						break;
-					}
-				}
+			if (
+				! blockBindings?.[ identifier ] ||
+				! canBindBlock( blockName )
+			) {
+				return {};
 			}
 
-			return _disableBoundBlocks;
+			const relevantBinding = blockBindings[ identifier ];
+			const { getBlockBindingsSource } = unlock( select( blocksStore ) );
+			const blockBindingsSource = getBlockBindingsSource(
+				relevantBinding.source
+			);
+
+			const _disableBoundBlock =
+				! blockBindingsSource?.canUserEditValue?.( {
+					select,
+					context: blockContext,
+					args: relevantBinding.args,
+				} );
+
+			const _bindingsPlaceholder = sprintf(
+				/* translators: %s: source label or key */
+				__( '%s value is empty' ),
```

Comment:
It should be a prompt for action, example:

```suggestion
				__( 'Type something to provide value for %s' ),
```

It should be rather more concise.

Reply:
There are two aspects to consider here:

* To simplify the workflows and the implementation, my idea was to reuse the same "empty" message when the user can edit and can not edit the meta field. For example, we are showing this message when you bind a paragraph to an empty field in a query loop, where editing is disabled.
* This message appears in every rich text, including shorter blocks like "Button". With that in mind, I would like to keep it as short as possible.

Do you think that showing " X value is empty" is not enough for prompting the user to write? Do you think that we should show different messages when the user can edit or can not?

Reply:
Yes, these are two different use cases. In particular, these placeholder prompts calling for action were designed with accessibility in mind. It's best to check it with assistive technology to check whether it makes sense.

Reply:
Okay, I have just pushed a commit to differentiate between these use cases. It is now a matter of deciding the text we want to show for each case. What about this:

Use "Add {key} value" when the user can edit:

![Screenshot 2024-08-30 at 15 05 52](https://github.com/user-attachments/assets/09184a27-d8d1-4a88-ac80-a4892858bdfe)

This follows what we are doing for post title or buttons for example. It is simple, but not sure if it is clear enough.

![Screenshot 2024-08-30 at 15 05 02](https://github.com/user-attachments/assets/32be86e7-9c5d-448b-8010-56018ef8d406)

Use just the key (or label) when the editing is locked.

![Screenshot 2024-08-30 at 15 06 15](https://github.com/user-attachments/assets/65da2c90-0cb5-4518-af84-bcd99dc700c4)

Reply:
Maybe just "Add empty_field"? Similar to add title, I think value is still not adding "value" 😆

Reply:
That could work for me 😄